pkg/orchestrator/data_scrubber.go

```diff
@@ -0,0 +1,127 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+// +build orchestrator
+
+package orchestrator
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/process/config"
+)
+
+var (
+	defaultSensitiveWords = []string{
+		"password", "passwd", "mysql_pwd",
+		"access_token", "auth_token",
+		"api_key", "apikey", "pwd",
+		"secret", "credentials", "stripetoken"}
+)
+
+// DataScrubber allows the agent to blacklist cmdline arguments that match
+// a list of predefined and custom words
+type DataScrubber struct {
+	Enabled                  bool
+	RegexSensitivePatterns   []*regexp.Regexp
+	LiteralSensitivePatterns []string
+	scrubbedCmdlines         map[string][]string
+}
+
+// NewDefaultDataScrubber creates a DataScrubber with the default behavior: enabled
+// and matching the default sensitive words
+func NewDefaultDataScrubber() *DataScrubber {
+	newDataScrubber := &DataScrubber{
+		Enabled:                  true,
+		LiteralSensitivePatterns: defaultSensitiveWords,
+		scrubbedCmdlines:         make(map[string][]string),
+	}
+
+	return newDataScrubber
+}
+func (ds *DataScrubber) ContainsBlacklistedWord(word string) bool {
+	for _, pattern := range ds.LiteralSensitivePatterns {
+		if strings.Contains(strings.ToLower(word), pattern) {
+			return true
+		}
+	}
+	return false
+}
+
+// ScrubCommand hides the argument value for any key which matches a "sensitive word" pattern.
+// It returns the updated cmdline, as well as a boolean representing whether it was scrubbed
+// future: we can add a check to do regex matching or simple matching depending whether we have RegexSensitivePatterns
+func (ds *DataScrubber) ScrubSimpleCommand(cmdline []string) ([]string, bool) {
+	changed := false
+	regexChanged := false
+	var wordReplacesIndexes []int
+
+	rawCmdline := strings.Join(cmdline, " ")
+	for _, pattern := range ds.RegexSensitivePatterns {
```

Comment:
if we add regex here we'll stop being faster than the original ScrubCommand right?

Reply:
On a high scale I suppose yes. But as long as we keep the number of regex low this should still be faster 💭

Reply:
ok cool, for now we don't need any regex if i understand correctly, so that's perfect. And this leaves us the option to add some later on if we realize we missed something so 💯

Reply:
Yes, this is only implemented as a function we have and not exposed as an option to the customer 👍